panic: nil pointer dereference in adkrest.NewServer when DebugConfig is not set

(I'm using google translate...)

  ## 🔴 Required Information                                                                                                                                                                                                        
                  
  **Describe the Bug:**
  `adkrest.NewServer` unconditionally dereferences `cfg.DebugConfig` (a pointer field)
  without a nil check. The built-in `api` sublauncher never sets `DebugConfig` when
  constructing `ServerConfig`, so using any launcher that includes `api` (e.g.
  `full.NewLauncher()`) always panics on startup.

  **Steps to Reproduce:**
  1. Install `google.golang.org/adk v1.1.0`
  2. Create a minimal agent using `full.NewLauncher()`
  3. Run with `web api webui` arguments
  4. Observe panic on startup (before the server begins listening)

  **Expected Behavior:**
  Server starts successfully. Missing `DebugConfig` should be treated as a zero value
  (defaulting to no custom trace capacity).

  **Observed Behavior:**
(I'm using vendor but not-vendor project occur same)
  ```text
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x7ff72b0dd52e]

goroutine 1 [running]:
google.golang.org/adk/server/adkrest.NewServer({{0x7ff72babe160, 0x2a50b0094360}, {0x0, 0x0}, {0x7ff72bab6c60, 0x2a50b00a6000}, {0x0, 0x0}, 0x1bf08eb000, {{0x0, ...}, ...}, ...})
        /vendor/google.golang.org/adk/server/adkrest/handler.go:39 +0x2e
google.golang.org/adk/cmd/launcher/web/api.(*apiLauncher).SetupSubrouters(0x2a50b009e0e0, 0x2a50b00ac000, 0x2a50b00a2090)
        /vendor/google.golang.org/adk/cmd/launcher/web/api/api.go:77 +0x14a
google.golang.org/adk/cmd/launcher/web.(*webLauncher).Run(0x2a50b0094270, {0x7ff72bab88e0, 0x7ff72c47c980}, 0x2a50b00a2090)
        /vendor/google.golang.org/adk/cmd/launcher/web/web.go:170 +0x45b
google.golang.org/adk/cmd/launcher/universal.(*uniLauncher).run(...)
        /vendor/google.golang.org/adk/cmd/launcher/universal/universal.go:65
google.golang.org/adk/cmd/launcher/universal.(*uniLauncher).ParseAndRun(0x2a50b00942a0, {0x7ff72bab88e0, 0x7ff72c47c980}, 0x2a50b00a2090, {0x2a50afdfe090?, 0x2a50b009e0a0?, 0x2a50b0094270?}, 0x7ff72ba9f580)
        /vendor/google.golang.org/adk/cmd/launcher/universal/universal.go:60 +0x7a
google.golang.org/adk/cmd/launcher/universal.(*uniLauncher).Execute(0x7ff72bac2860?, {0x7ff72bab88e0?, 0x7ff72c47c980?}, 0x0?, {0x2a50afdfe090?, 0x0?, 0x8?})
        /vendor/google.golang.org/adk/cmd/launcher/universal/universal.go:36 +0x29
main.main()
        /agent.go:229 +0x488
exit status 2
```

  Environment Details:
  - ADK Library Version: v1.1.0
  - OS: Windows 10
  - Go Version: 1.23

  Model Information:
  - N/A (panic occurs before any model is invoked)

  ---
  🟡 Optional Information

  Code: (Same as sample code in adk document)
```
package main

import (
    "context"
    "log"
    "os"

    "google.golang.org/adk/agent"
    "google.golang.org/adk/agent/llmagent"
    "google.golang.org/adk/cmd/launcher"
    "google.golang.org/adk/cmd/launcher/full"
    "google.golang.org/adk/model/gemini"
    "google.golang.org/adk/tool"
    "google.golang.org/adk/tool/geminitool"
    "google.golang.org/genai"
)

func main() {
    ctx := context.Background()

    model, err := gemini.NewModel(ctx, "gemini-flash-latest", &genai.ClientConfig{
        APIKey: os.Getenv("GOOGLE_API_KEY"),
    })
    if err != nil {
        log.Fatalf("Failed to create model: %v", err)
    }

    timeAgent, err := llmagent.New(llmagent.Config{
        Name:        "hello_time_agent",
        Model:       model,
        Description: "Tells the current time in a specified city.",
        Instruction: "You are a helpful assistant that tells the current time in a city.",
        Tools: []tool.Tool{
            geminitool.GoogleSearch{},
        },
    })
    if err != nil {
        log.Fatalf("Failed to create agent: %v", err)
    }

    config := &launcher.Config{
        AgentLoader: agent.NewSingleLoader(timeAgent),
    }

    l := full.NewLauncher()
    if err = l.Execute(ctx, config, os.Args[1:]); err != nil {
        log.Fatalf("Run failed: %v\n\n%s", err, l.CommandLineSyntax())
    }
}
```

 Root Cause
```

  In server/adkrest/handler.go:39:

  // DebugConfig is *DebugTelemetryConfig (a pointer), but is never set by the
  // api sublauncher (api.go:77-84), so it is always nil here.
  debugTelemetry, err := services.NewDebugTelemetryWithConfig(&services.DebugTelemetryConfig{
      TraceCapacity: cfg.DebugConfig.TraceCapacity, // nil dereference
  })
```

  api.go constructs ServerConfig without setting DebugConfig, leaving it nil.

  Suggested Fix (Gemini suggested, and it works)
```

  var debugTraceCapacity int
  if cfg.DebugConfig != nil {
      debugTraceCapacity = cfg.DebugConfig.TraceCapacity
  }
  debugTelemetry, err := services.NewDebugTelemetryWithConfig(&services.DebugTelemetryConfig{
      TraceCapacity: debugTraceCapacity,
  })
```

  How often has this issue occurred?:
  - Always (100%)

--- a/cmd/launcher/web/api/api.go
+++ b/cmd/launcher/web/api/api.go
@@ -36,6 +36,7 @@ type apiConfig struct {
 	frontendAddress string
 	pathPrefix      string
 	sseWriteTimeout time.Duration
+	traceCapacity   int
 }
 
 // apiLauncher can launch ADK REST API
@@ -81,6 +82,9 @@ func (a *apiLauncher) SetupSubrouters(router *mux.Router, config *launcher.Confi
 		ArtifactService: config.ArtifactService,
 		SSEWriteTimeout: a.config.sseWriteTimeout,
 		PluginConfig:    config.PluginConfig,
+		DebugConfig: &adkrest.DebugTelemetryConfig{
+			TraceCapacity: a.config.traceCapacity,
+		},
 	})
 	if err != nil {
 		return fmt.Errorf("failed to create REST server: %w", err)
@@ -138,6 +142,7 @@ func NewLauncher() weblauncher.Sublauncher {
 	fs.StringVar(&config.frontendAddress, "webui_address", "localhost:8080", "ADK WebUI address as seen from the user browser. It's used to allow CORS requests. Please specify only hostname and (optionally) port.")
 	fs.StringVar(&config.pathPrefix, "path_prefix", "/api", "ADK REST API path prefix. Default is '/api'.")
 	fs.DurationVar(&config.sseWriteTimeout, "sse-write-timeout", 120*time.Second, "SSE server write timeout (i.e. '10s', '2m' - see time.ParseDuration for details) - for writing the SSE response after reading the headers & body")
+	fs.IntVar(&config.traceCapacity, "trace_capacity", 1000, "Maximum number of traces to keep in memory.")
 
 	return &apiLauncher{
 		config: config,

--- a/cmd/launcher/web/api/api.go
+++ b/cmd/launcher/web/api/api.go
@@ -82,7 +82,7 @@ func (a *apiLauncher) SetupSubrouters(router *mux.Router, config *launcher.Confi
 		ArtifactService: config.ArtifactService,
 		SSEWriteTimeout: a.config.sseWriteTimeout,
 		PluginConfig:    config.PluginConfig,
-		DebugConfig: &adkrest.DebugTelemetryConfig{
+		DebugConfig: adkrest.DebugTelemetryConfig{
 			TraceCapacity: a.config.traceCapacity,
 		},
 	})

--- a/cmd/launcher/web/api/api.go
+++ b/cmd/launcher/web/api/api.go
@@ -142,7 +142,7 @@ func NewLauncher() weblauncher.Sublauncher {
 	fs.StringVar(&config.frontendAddress, "webui_address", "localhost:8080", "ADK WebUI address as seen from the user browser. It's used to allow CORS requests. Please specify only hostname and (optionally) port.")
 	fs.StringVar(&config.pathPrefix, "path_prefix", "/api", "ADK REST API path prefix. Default is '/api'.")
 	fs.DurationVar(&config.sseWriteTimeout, "sse-write-timeout", 120*time.Second, "SSE server write timeout (i.e. '10s', '2m' - see time.ParseDuration for details) - for writing the SSE response after reading the headers & body")
-	fs.IntVar(&config.traceCapacity, "trace_capacity", 1000, "Maximum number of traces to keep in memory.")
+	fs.IntVar(&config.traceCapacity, "trace_capacity", 10000, "Maximum number of traces to keep in memory.")
 
 	return &apiLauncher{
 		config: config,

--- a/server/adkrest/handler.go
+++ b/server/adkrest/handler.go
@@ -67,7 +67,7 @@ type ServerConfig struct {
 	ArtifactService artifact.Service
 	SSEWriteTimeout time.Duration
 	PluginConfig    runner.PluginConfig
-	DebugConfig     *DebugTelemetryConfig
+	DebugConfig     DebugTelemetryConfig
 }
 
 // DebugTelemetryConfig contains parameters for the debug telemetry.

